--- DSDT_RPI_7c51feb.dsl	2025-10-31 12:34:36.745308000 -0400
+++ DSDT_RPI_2cab24f.dsl	2025-10-31 12:34:34.071491000 -0400
@@ -1,4 +1,4 @@
-DefinitionBlock ("", "DSDT", 5, "RPIFDN", "RPI", 0x00000002)
+DefinitionBlock ("", "DSDT", 2, "RPIFDN", "RPI", 0x00000002)
 {
     Scope (_SB)
     {
@@ -74,85 +74,6 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             })
         }
 
-        Device (SCB0)
-        {
-            Name (_HID, "ACPI0004" /* Module Device */)  // _HID: Hardware ID
-            Name (_UID, Zero)  // _UID: Unique ID
-            Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
-            Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
-            {
-                Name (RBUF, ResourceTemplate ()
-                {
-                    QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
-                        0x0000000000000000, // Granularity
-                        0x0000000600000000, // Range Minimum
-                        0x0000000600000000, // Range Maximum
-                        0x0000000000000000, // Translation Offset
-                        0x0000000000000001, // Length
-                        ,, _Y00, AddressRangeMemory, TypeStatic)
-                })
-                CreateQWordField (RBUF, \_SB.SCB0._CRS._Y00._MAX, MMBE)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.SCB0._CRS._Y00._LEN, MMLE)  // _LEN: Length
-                MMBE += 0x0FFF
-                MMLE += 0x0FFF
-                Return (RBUF) /* \_SB_.SCB0._CRS.RBUF */
-            }
-
-            Name (_DMA, ResourceTemplate ()  // _DMA: Direct Memory Access
-            {
-                QWordMemory (ResourceConsumer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
-                    0x0000000000000000, // Granularity
-                    0x0000000000000000, // Range Minimum
-                    0x00000000BFFFFFFF, // Range Maximum
-                    0x0000000000000000, // Translation Offset
-                    0x00000000C0000000, // Length
-                    ,, , AddressRangeMemory, TypeStatic)
-            })
-            Device (XHC0)
-            {
-                Name (_HID, "PNP0D10" /* XHCI USB Controller with debug */)  // _HID: Hardware ID
-                Name (_UID, Zero)  // _UID: Unique ID
-                Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
-                Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
-                {
-                    Name (RBUF, ResourceTemplate ()
-                    {
-                        QWordMemory (ResourceConsumer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
-                            0x0000000000000000, // Granularity
-                            0x0000000600000000, // Range Minimum
-                            0x0000000600000000, // Range Maximum
-                            0x0000000000000000, // Translation Offset
-                            0x0000000000000001, // Length
-                            ,, _Y01, AddressRangeMemory, TypeStatic)
-                        Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
-                        {
-                            0x000000AF,
-                        }
-                    })
-                    CreateQWordField (RBUF, \_SB.SCB0.XHC0._CRS._Y01._MAX, MMBE)  // _MAX: Maximum Base Address
-                    CreateQWordField (RBUF, \_SB.SCB0.XHC0._CRS._Y01._LEN, MMLE)  // _LEN: Length
-                    MMBE += 0x0FFF
-                    MMLE += 0x0FFF
-                    Return (RBUF) /* \_SB_.SCB0.XHC0._CRS.RBUF */
-                }
-
-                Method (_INI, 0, Serialized)  // _INI: Initialize
-                {
-                    OperationRegion (PCFG, SystemMemory, 0xFD508000, 0x1000)
-                    Field (PCFG, AnyAcc, NoLock, Preserve)
-                    {
-                        VNID,   16, 
-                        DVID,   16, 
-                        CMND,   16, 
-                        STAT,   16
-                    }
-
-                    Debug = "xHCI enable"
-                    CMND = 0x06
-                }
-            }
-        }
-
         Device (CPU0)
         {
             Name (_HID, "ACPI0007" /* Processor Device */)  // _HID: Hardware ID
@@ -208,6 +129,20 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                         0x0000000000000000, // Range Maximum
                         0x0000000000000000, // Translation Offset
                         0x0000000000000001, // Length
+                        ,, _Y00, AddressRangeMemory, TypeStatic)
+                    QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
+                        0x0000000000000000, // Granularity
+                        0x0000000000000000, // Range Minimum
+                        0x0000000000000000, // Range Maximum
+                        0x0000000000000000, // Translation Offset
+                        0x0000000000000001, // Length
+                        ,, _Y01, AddressRangeMemory, TypeStatic)
+                    QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
+                        0x0000000000000000, // Granularity
+                        0x0000000000000000, // Range Minimum
+                        0x0000000000000000, // Range Maximum
+                        0x0000000000000000, // Translation Offset
+                        0x0000000000000001, // Length
                         ,, _Y02, AddressRangeMemory, TypeStatic)
                     QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
                         0x0000000000000000, // Granularity
@@ -342,150 +277,136 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                         0x0000000000000000, // Translation Offset
                         0x0000000000000001, // Length
                         ,, _Y15, AddressRangeMemory, TypeStatic)
-                    QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
-                        0x0000000000000000, // Granularity
-                        0x0000000000000000, // Range Minimum
-                        0x0000000000000000, // Range Maximum
-                        0x0000000000000000, // Translation Offset
-                        0x0000000000000001, // Length
-                        ,, _Y16, AddressRangeMemory, TypeStatic)
-                    QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
-                        0x0000000000000000, // Granularity
-                        0x0000000000000000, // Range Minimum
-                        0x0000000000000000, // Range Maximum
-                        0x0000000000000000, // Translation Offset
-                        0x0000000000000001, // Length
-                        ,, _Y17, AddressRangeMemory, TypeStatic)
                 })
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._MIN, MI01)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._MAX, MA01)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._LEN, LE01)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y00._MIN, MI01)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y00._MAX, MA01)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y00._LEN, LE01)  // _LEN: Length
                 LE01 = 0x00010000
                 MI01 = 0xFE980000
                 MA01 = (MI01 + (LE01 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._MIN, MI02)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._MAX, MA02)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._LEN, LE02)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y01._MIN, MI02)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y01._MAX, MA02)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y01._LEN, LE02)  // _LEN: Length
                 LE02 = 0x1000
                 MI02 = 0xFEC00000
                 MA02 = (MI02 + (LE02 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._MIN, MI03)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._MAX, MA03)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._LEN, LE03)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._MIN, MI03)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._MAX, MA03)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._LEN, LE03)  // _LEN: Length
                 LE03 = 0x6000
                 MI03 = 0xFE400000
                 MA03 = (MI03 + (LE03 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._MIN, MI06)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._MAX, MA06)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._LEN, LE06)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._MIN, MI06)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._MAX, MA06)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._LEN, LE06)  // _LEN: Length
                 LE06 = 0x0100
                 MI06 = 0xFE807000
                 MA06 = (MI06 + (LE06 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._MIN, MI07)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._MAX, MA07)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._LEN, LE07)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._MIN, MI07)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._MAX, MA07)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._LEN, LE07)  // _LEN: Length
                 LE07 = 0x0600
                 MI07 = 0xFE902000
                 MA07 = (MI07 + (LE07 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._MIN, MI08)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._MAX, MA08)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._LEN, LE08)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._MIN, MI08)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._MAX, MA08)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._LEN, LE08)  // _LEN: Length
                 LE08 = 0x0100
                 MI08 = 0xFE808000
                 MA08 = (MI08 + (LE08 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._MIN, MI09)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._MAX, MA09)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._LEN, LE09)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._MIN, MI09)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._MAX, MA09)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._LEN, LE09)  // _LEN: Length
                 LE09 = 0x24
                 MI09 = 0xFE00B880
                 MA09 = (MI09 + (LE09 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._MIN, MI10)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._MAX, MA10)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._LEN, LE10)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._MIN, MI10)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._MAX, MA10)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._LEN, LE10)  // _LEN: Length
                 LE10 = 0x10
                 MI10 = 0xFE00B840
                 MA10 = (MI10 + (LE10 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._MIN, MI11)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._MAX, MA11)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._LEN, LE11)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._MIN, MI11)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._MAX, MA11)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._LEN, LE11)  // _LEN: Length
                 LE11 = 0xB4
                 MI11 = 0xFE200000
                 MA11 = (MI11 + (LE11 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._MIN, MI12)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._MAX, MA12)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._LEN, LE12)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._MIN, MI12)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._MAX, MA12)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._LEN, LE12)  // _LEN: Length
                 LE12 = 0x20
                 MI12 = 0xFE804000
                 MA12 = (MI12 + (LE12 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._MIN, MI13)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._MAX, MA13)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._LEN, LE13)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._MIN, MI13)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._MAX, MA13)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._LEN, LE13)  // _LEN: Length
                 LE13 = 0x20
                 MI13 = 0xFE805000
                 MA13 = (MI13 + (LE13 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._MIN, MI14)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._MAX, MA14)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._LEN, LE14)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._MIN, MI14)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._MAX, MA14)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._LEN, LE14)  // _LEN: Length
                 LE14 = 0x20
                 MI14 = 0xFE204000
                 MA14 = (MI14 + (LE14 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._MIN, MI15)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._MAX, MA15)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._LEN, LE15)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._MIN, MI15)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._MAX, MA15)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._LEN, LE15)  // _LEN: Length
                 LE15 = 0x40
                 MI15 = 0xFE215080
                 MA15 = (MI15 + (LE15 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._MIN, MI17)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._MAX, MA17)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._LEN, LE17)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._MIN, MI17)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._MAX, MA17)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._LEN, LE17)  // _LEN: Length
                 LE17 = 0x0100
                 MI17 = 0xFE007B00
                 MA17 = (MI17 + (LE17 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._MIN, MI18)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._MAX, MA18)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._LEN, LE18)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._MIN, MI18)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._MAX, MA18)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._LEN, LE18)  // _LEN: Length
                 LE18 = 0x28
                 MI18 = 0xFE20C000
                 MA18 = (MI18 + (LE18 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._MIN, MI19)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._MAX, MA19)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._LEN, LE19)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._MIN, MI19)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._MAX, MA19)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._LEN, LE19)  // _LEN: Length
                 MI19 = 0x7E20C000
                 LE19 = 0x28
                 MA19 = (MI19 + (LE19 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._MIN, MI20)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._MAX, MA20)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._LEN, LE20)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._MIN, MI20)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._MAX, MA20)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._LEN, LE20)  // _LEN: Length
                 MI20 = 0xFF20C000
                 LE20 = 0x28
                 MA20 = (MI20 + (LE20 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._MIN, MI21)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._MAX, MA21)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._LEN, LE21)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._MIN, MI21)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._MAX, MA21)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._LEN, LE21)  // _LEN: Length
                 LE21 = 0x08
                 MI21 = 0xFE1010A0
                 MA21 = (MI21 + (LE21 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._MIN, MI22)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._MAX, MA22)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._LEN, LE22)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._MIN, MI22)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._MAX, MA22)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._LEN, LE22)  // _LEN: Length
                 LE22 = 0x1000
                 MI22 = 0xFE201000
                 MA22 = (MI22 + (LE22 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._MIN, MI23)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._MAX, MA23)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._LEN, LE23)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._MIN, MI23)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._MAX, MA23)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._LEN, LE23)  // _LEN: Length
                 LE23 = 0x70
                 MI23 = 0xFE215000
                 MA23 = (MI23 + (LE23 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y16._MIN, MI24)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y16._MAX, MA24)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y16._LEN, LE24)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._MIN, MI24)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._MAX, MA24)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._LEN, LE24)  // _LEN: Length
                 LE24 = 0x0100
                 MI24 = 0xFE300000
                 MA24 = (MI24 + (LE24 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y17._MIN, MI25)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y17._MAX, MA25)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y17._LEN, LE25)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._MIN, MI25)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._MAX, MA25)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._LEN, LE25)  // _LEN: Length
                 LE25 = 0x0100
                 MI25 = 0xFE202000
                 MA25 = (MI25 + (LE25 - One))
@@ -494,7 +415,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
 
             Name (_DMA, ResourceTemplate ()  // _DMA: Direct Memory Access
             {
-                QWordMemory (ResourceConsumer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
+                QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
                     0x0000000000000000, // Granularity
                     0x00000000C0000000, // Range Minimum
                     0x00000000FFFFFFFF, // Range Maximum
@@ -505,11 +426,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (USB0)
             {
                 Name (_HID, "BCM2848")  // _HID: Hardware ID
-                Name (_CID, Package (0x02)  // _CID: Compatible ID
-                {
-                    "DWC_OTG", 
-                    "DWC2_OTG"
-                })
+                Name (_CID, "BCM2848")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -522,7 +439,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00010000,         // Address Length
-                        _Y18)
+                        _Y16)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x00000069,
@@ -530,7 +447,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.USB0._Y18._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.USB0._Y16._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE980000
                     Return (RBUF) /* \_SB_.GDV0.USB0.RBUF */
                 }
@@ -539,7 +456,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (GPU0)
             {
                 Name (_HID, "BCM2850")  // _HID: Hardware ID
-                Name (_CID, "VC4")  // _CID: Compatible ID
+                Name (_CID, "BCM2850")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -552,7 +469,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00001000,         // Address Length
-                        _Y19)
+                        _Y17)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x0000002A,
@@ -560,11 +477,11 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00006000,         // Address Length
-                        _Y1A)
+                        _Y18)
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000100,         // Address Length
-                        _Y1B)
+                        _Y19)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x0000004A,
@@ -572,11 +489,11 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000600,         // Address Length
-                        _Y1C)
+                        _Y1A)
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000100,         // Address Length
-                        _Y1D)
+                        _Y1B)
                     I2cSerialBusV2 (0x0050, ControllerInitiated, 0x000186A0,
                         AddressingMode7Bit, "\\_SB.GDV0.I2C2",
                         0x00, ResourceConsumer, , Exclusive,
@@ -588,15 +505,15 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y19._BAS, RB01)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y17._BAS, RB01)  // _BAS: Base Address
                     RB01 = 0xFEC00000
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1A._BAS, RB02)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y18._BAS, RB02)  // _BAS: Base Address
                     RB02 = 0xFE400000
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1B._BAS, RB05)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y19._BAS, RB05)  // _BAS: Base Address
                     RB05 = 0xFE807000
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1C._BAS, RB06)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1A._BAS, RB06)  // _BAS: Base Address
                     RB06 = 0xFE902000
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1D._BAS, RB07)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1B._BAS, RB07)  // _BAS: Base Address
                     RB07 = 0xFE808000
                     Return (RBUF) /* \_SB_.GDV0.GPU0.RBUF */
                 }
@@ -643,7 +560,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (RPIQ)
             {
                 Name (_HID, "BCM2849")  // _HID: Hardware ID
-                Name (_CID, "RPIQ")  // _CID: Compatible ID
+                Name (_CID, "BCM2849")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -656,7 +573,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000024,         // Address Length
-                        _Y1E)
+                        _Y1C)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x00000041,
@@ -664,7 +581,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.RPIQ._Y1E._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.RPIQ._Y1C._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE00B880
                     Return (RBUF) /* \_SB_.GDV0.RPIQ.RBUF */
                 }
@@ -673,7 +590,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (VCIQ)
             {
                 Name (_HID, "BCM2835")  // _HID: Hardware ID
-                Name (_CID, "VCIQ")  // _CID: Compatible ID
+                Name (_CID, "BCM2835")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Name (_DEP, Package (0x01)  // _DEP: Dependencies
@@ -690,7 +607,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000010,         // Address Length
-                        _Y1F)
+                        _Y1D)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x00000042,
@@ -698,7 +615,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.VCIQ._Y1F._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.VCIQ._Y1D._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE00B840
                     Return (RBUF) /* \_SB_.GDV0.VCIQ.RBUF */
                 }
@@ -707,7 +624,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (VCSM)
             {
                 Name (_HID, "BCM2856")  // _HID: Hardware ID
-                Name (_CID, "VCSM")  // _CID: Compatible ID
+                Name (_CID, "BCM2856")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Name (_DEP, Package (0x01)  // _DEP: Dependencies
@@ -723,7 +640,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (GPI0)
             {
                 Name (_HID, "BCM2845")  // _HID: Hardware ID
-                Name (_CID, "BCMGPIO")  // _CID: Compatible ID
+                Name (_CID, "BCM2845")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -736,7 +653,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x000000B4,         // Address Length
-                        _Y20)
+                        _Y1E)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x00000091,
@@ -747,7 +664,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.GPI0._Y20._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPI0._Y1E._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE200000
                     Return (RBUF) /* \_SB_.GDV0.GPI0.RBUF */
                 }
@@ -756,7 +673,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (I2C1)
             {
                 Name (_HID, "BCM2841")  // _HID: Hardware ID
-                Name (_CID, "BCMI2C")  // _CID: Compatible ID
+                Name (_CID, "BCM2841")  // _CID: Compatible ID
                 Name (_UID, One)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -769,7 +686,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000020,         // Address Length
-                        _Y21)
+                        _Y1F)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x00000095,
@@ -783,7 +700,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.I2C1._Y21._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.I2C1._Y1F._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE804000
                     Return (RBUF) /* \_SB_.GDV0.I2C1.RBUF */
                 }
@@ -792,7 +709,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (I2C2)
             {
                 Name (_HID, "BCM2841")  // _HID: Hardware ID
-                Name (_CID, "BCMI2C")  // _CID: Compatible ID
+                Name (_CID, "BCM2841")  // _CID: Compatible ID
                 Name (_UID, 0x02)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -805,7 +722,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000020,         // Address Length
-                        _Y22)
+                        _Y20)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x00000095,
@@ -813,7 +730,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.I2C2._Y22._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.I2C2._Y20._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE805000
                     Return (RBUF) /* \_SB_.GDV0.I2C2.RBUF */
                 }
@@ -822,7 +739,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (SPI0)
             {
                 Name (_HID, "BCM2838")  // _HID: Hardware ID
-                Name (_CID, "BCMSPI0")  // _CID: Compatible ID
+                Name (_CID, "BCM2838")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -835,7 +752,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000020,         // Address Length
-                        _Y23)
+                        _Y21)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x00000096,
@@ -860,7 +777,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.SPI0._Y23._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.SPI0._Y21._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE204000
                     Return (RBUF) /* \_SB_.GDV0.SPI0.RBUF */
                 }
@@ -869,7 +786,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (SPI1)
             {
                 Name (_HID, "BCM2839")  // _HID: Hardware ID
-                Name (_CID, "BCMAUXSPI")  // _CID: Compatible ID
+                Name (_CID, "BCM2839")  // _CID: Compatible ID
                 Name (_UID, One)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Name (_DEP, Package (0x01)  // _DEP: Dependencies
@@ -886,7 +803,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000040,         // Address Length
-                        _Y24)
+                        _Y22)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x0000007D,
@@ -906,7 +823,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.SPI1._Y24._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.SPI1._Y22._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE215080
                     Return (RBUF) /* \_SB_.GDV0.SPI1.RBUF */
                 }
@@ -928,11 +845,11 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000100,         // Address Length
-                        _Y25)
+                        _Y23)
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000028,         // Address Length
-                        _Y26)
+                        _Y24)
                     Memory32Fixed (ReadWrite,
                         0x7E20C000,         // Address Base
                         0x00000028,         // Address Length
@@ -944,7 +861,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000008,         // Address Length
-                        _Y27)
+                        _Y25)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x0000003B,
@@ -953,11 +870,11 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.PWM0._Y25._BAS, RB01)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.PWM0._Y23._BAS, RB01)  // _BAS: Base Address
                     RB01 = 0xFE007B00
-                    CreateDWordField (RBUF, \_SB.GDV0.PWM0._Y26._BAS, RB02)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.PWM0._Y24._BAS, RB02)  // _BAS: Base Address
                     RB02 = 0xFE20C000
-                    CreateDWordField (RBUF, \_SB.GDV0.PWM0._Y27._BAS, RB03)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.PWM0._Y25._BAS, RB03)  // _BAS: Base Address
                     RB03 = 0xFE1010A0
                     Return (RBUF) /* \_SB_.GDV0.PWM0.RBUF */
                 }
@@ -979,7 +896,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00001000,         // Address Length
-                        _Y28)
+                        _Y26)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x00000099,
@@ -987,7 +904,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.URT0._Y28._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.URT0._Y26._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE201000
                     Return (RBUF) /* \_SB_.GDV0.URT0.RBUF */
                 }
@@ -1010,7 +927,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (URTM)
             {
                 Name (_HID, "BCM2836")  // _HID: Hardware ID
-                Name (_CID, "MINIUART")  // _CID: Compatible ID
+                Name (_CID, "BCM2836")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -1023,7 +940,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000070,         // Address Length
-                        _Y29)
+                        _Y27)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x0000007D,
@@ -1031,38 +948,62 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.URTM._Y29._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.URTM._Y27._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE215000
                     Return (RBUF) /* \_SB_.GDV0.URTM.RBUF */
                 }
+
+                Name (MUCR, 0x01000000)
+                Name (_DSD, Package (0x02)  // _DSD: Device-Specific Data
+                {
+                    ToUUID ("daffd814-6eba-4d8c-8a91-bc9bbf4aa301") /* Device Properties for _DSD */, 
+                    Package (0x01)
+                    {
+                        Package (0x02)
+                        {
+                            "clock-frequency", 
+                            MUCR
+                        }
+                    }
+                })
             }
 
             Device (BTH0)
             {
                 Name (_HID, "BCM2EA6")  // _HID: Hardware ID
                 Name (_CID, "BCM2EA6")  // _CID: Compatible ID
+                Name (URIU, 0x02)
                 Method (_STA, 0, NotSerialized)  // _STA: Status
                 {
                     Return (0x0F)
                 }
 
+                Name (BTPL, ResourceTemplate ()
+                {
+                    UartSerialBusV2 (0x0001C200, DataBitsEight, StopBitsOne,
+                        0x00, LittleEndian, ParityTypeNone, FlowControlNone,
+                        0x0010, 0x0010, "\\_SB.GDV0.URT0",
+                        0x00, ResourceConsumer, , Exclusive,
+                        )
+                })
+                Name (BTMN, ResourceTemplate ()
+                {
+                    UartSerialBusV2 (0x0001C200, DataBitsEight, StopBitsOne,
+                        0x00, LittleEndian, ParityTypeNone, FlowControlNone,
+                        0x0010, 0x0010, "\\_SB.GDV0.URTM",
+                        0x00, ResourceConsumer, , Exclusive,
+                        )
+                })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    Name (RBUF, ResourceTemplate ()
+                    If ((URIU == Zero))
                     {
-                        UartSerialBusV2 (0x0001C200, DataBitsEight, StopBitsOne,
-                            0x00, LittleEndian, ParityTypeNone, FlowControlNone,
-                            0x0010, 0x0010, "\\_SB.GDV0.URTM",
-                            0x00, ResourceConsumer, , Exclusive,
-                            )
-                        GpioIo (Shared, PullUp, 0x0000, 0x0000, IoRestrictionNone,
-                            "\\_SB.GDV0.RPIQ", 0x00, ResourceConsumer, ,
-                            )
-                            {   // Pin list
-                                0x0080
-                            }
-                    })
-                    Return (RBUF) /* \_SB_.GDV0.BTH0._CRS.RBUF */
+                        Return (BTMN) /* \_SB_.GDV0.BTH0.BTMN */
+                    }
+                    Else
+                    {
+                        Return (BTPL) /* \_SB_.GDV0.BTH0.BTPL */
+                    }
                 }
             }
 
@@ -1480,7 +1421,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (SDC1)
             {
                 Name (_HID, "BCM2847")  // _HID: Hardware ID
-                Name (_CID, "ARASAN")  // _CID: Compatible ID
+                Name (_CID, "BCM2847")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Name (_S1D, One)  // _S1D: S1 Device State
@@ -1497,7 +1438,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000100,         // Address Length
-                        _Y2A)
+                        _Y28)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x0000009E,
@@ -1505,7 +1446,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.SDC1._Y2A._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.SDC1._Y28._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE300000
                     Return (RBUF) /* \_SB_.GDV0.SDC1.RBUF */
                 }
@@ -1518,7 +1459,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                         Package (0x02)
                         {
                             "sdhci-caps", 
-                            0x0100FA81
+                            0x0120FA81
                         }
                     }
                 })
