--- DSDT_v0.8...v0.11.dsl	2025-10-31 12:04:49.496283700 -0400
+++ DSDT_v0.16...v0.17.dsl	2025-10-31 12:04:49.448980700 -0400
@@ -1,4 +1,4 @@
-DefinitionBlock ("", "DSDT", 5, "RPIFDN", "RPI", 0x00000002)
+DefinitionBlock ("", "DSDT", 2, "RPIFDN", "RPI", 0x00000002)
 {
     Scope (_SB)
     {
@@ -74,85 +74,6 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             })
         }
 
-        Device (SCB0)
-        {
-            Name (_HID, "ACPI0004" /* Module Device */)  // _HID: Hardware ID
-            Name (_UID, Zero)  // _UID: Unique ID
-            Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
-            Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
-            {
-                Name (RBUF, ResourceTemplate ()
-                {
-                    QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
-                        0x0000000000000000, // Granularity
-                        0x0000000600000000, // Range Minimum
-                        0x0000000600000000, // Range Maximum
-                        0x0000000000000000, // Translation Offset
-                        0x0000000000000001, // Length
-                        ,, _Y00, AddressRangeMemory, TypeStatic)
-                })
-                CreateQWordField (RBUF, \_SB.SCB0._CRS._Y00._MAX, MMBE)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.SCB0._CRS._Y00._LEN, MMLE)  // _LEN: Length
-                MMBE += 0x0FFF
-                MMLE += 0x0FFF
-                Return (RBUF) /* \_SB_.SCB0._CRS.RBUF */
-            }
-
-            Name (_DMA, ResourceTemplate ()  // _DMA: Direct Memory Access
-            {
-                QWordMemory (ResourceConsumer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
-                    0x0000000000000000, // Granularity
-                    0x0000000000000000, // Range Minimum
-                    0x00000000BFFFFFFF, // Range Maximum
-                    0x0000000000000000, // Translation Offset
-                    0x00000000C0000000, // Length
-                    ,, , AddressRangeMemory, TypeStatic)
-            })
-            Device (XHC0)
-            {
-                Name (_HID, "PNP0D10" /* XHCI USB Controller with debug */)  // _HID: Hardware ID
-                Name (_UID, Zero)  // _UID: Unique ID
-                Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
-                Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
-                {
-                    Name (RBUF, ResourceTemplate ()
-                    {
-                        QWordMemory (ResourceConsumer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
-                            0x0000000000000000, // Granularity
-                            0x0000000600000000, // Range Minimum
-                            0x0000000600000000, // Range Maximum
-                            0x0000000000000000, // Translation Offset
-                            0x0000000000000001, // Length
-                            ,, _Y01, AddressRangeMemory, TypeStatic)
-                        Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
-                        {
-                            0x000000AF,
-                        }
-                    })
-                    CreateQWordField (RBUF, \_SB.SCB0.XHC0._CRS._Y01._MAX, MMBE)  // _MAX: Maximum Base Address
-                    CreateQWordField (RBUF, \_SB.SCB0.XHC0._CRS._Y01._LEN, MMLE)  // _LEN: Length
-                    MMBE += 0x0FFF
-                    MMLE += 0x0FFF
-                    Return (RBUF) /* \_SB_.SCB0.XHC0._CRS.RBUF */
-                }
-
-                Method (_INI, 0, Serialized)  // _INI: Initialize
-                {
-                    OperationRegion (PCFG, SystemMemory, 0xFD508000, 0x1000)
-                    Field (PCFG, AnyAcc, NoLock, Preserve)
-                    {
-                        VNID,   16, 
-                        DVID,   16, 
-                        CMND,   16, 
-                        STAT,   16
-                    }
-
-                    Debug = "xHCI enable"
-                    CMND = 0x06
-                }
-            }
-        }
-
         Device (CPU0)
         {
             Name (_HID, "ACPI0007" /* Processor Device */)  // _HID: Hardware ID
@@ -208,6 +129,20 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                         0x0000000000000000, // Range Maximum
                         0x0000000000000000, // Translation Offset
                         0x0000000000000001, // Length
+                        ,, _Y00, AddressRangeMemory, TypeStatic)
+                    QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
+                        0x0000000000000000, // Granularity
+                        0x0000000000000000, // Range Minimum
+                        0x0000000000000000, // Range Maximum
+                        0x0000000000000000, // Translation Offset
+                        0x0000000000000001, // Length
+                        ,, _Y01, AddressRangeMemory, TypeStatic)
+                    QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
+                        0x0000000000000000, // Granularity
+                        0x0000000000000000, // Range Minimum
+                        0x0000000000000000, // Range Maximum
+                        0x0000000000000000, // Translation Offset
+                        0x0000000000000001, // Length
                         ,, _Y02, AddressRangeMemory, TypeStatic)
                     QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
                         0x0000000000000000, // Granularity
@@ -349,152 +284,151 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                         0x0000000000000000, // Translation Offset
                         0x0000000000000001, // Length
                         ,, _Y16, AddressRangeMemory, TypeStatic)
-                    QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
-                        0x0000000000000000, // Granularity
-                        0x0000000000000000, // Range Minimum
-                        0x0000000000000000, // Range Maximum
-                        0x0000000000000000, // Translation Offset
-                        0x0000000000000001, // Length
-                        ,, _Y17, AddressRangeMemory, TypeStatic)
                 })
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._MIN, MI01)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._MAX, MA01)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._LEN, LE01)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y00._MIN, MI01)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y00._MAX, MA01)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y00._LEN, LE01)  // _LEN: Length
                 LE01 = 0x00010000
                 MI01 = 0xFE980000
                 MA01 = (MI01 + (LE01 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._MIN, MI02)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._MAX, MA02)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._LEN, LE02)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y01._MIN, MI02)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y01._MAX, MA02)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y01._LEN, LE02)  // _LEN: Length
                 LE02 = 0x1000
                 MI02 = 0xFEC00000
                 MA02 = (MI02 + (LE02 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._MIN, MI03)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._MAX, MA03)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._LEN, LE03)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._MIN, MI03)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._MAX, MA03)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y02._LEN, LE03)  // _LEN: Length
                 LE03 = 0x6000
                 MI03 = 0xFE400000
                 MA03 = (MI03 + (LE03 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._MIN, MI06)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._MAX, MA06)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._LEN, LE06)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._MIN, MI06)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._MAX, MA06)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y03._LEN, LE06)  // _LEN: Length
                 LE06 = 0x0100
                 MI06 = 0xFE807000
                 MA06 = (MI06 + (LE06 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._MIN, MI07)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._MAX, MA07)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._LEN, LE07)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._MIN, MI07)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._MAX, MA07)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y04._LEN, LE07)  // _LEN: Length
                 LE07 = 0x0600
                 MI07 = 0xFE902000
                 MA07 = (MI07 + (LE07 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._MIN, MI08)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._MAX, MA08)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._LEN, LE08)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._MIN, MI08)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._MAX, MA08)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y05._LEN, LE08)  // _LEN: Length
                 LE08 = 0x0100
                 MI08 = 0xFE808000
                 MA08 = (MI08 + (LE08 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._MIN, MI09)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._MAX, MA09)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._LEN, LE09)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._MIN, MI09)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._MAX, MA09)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y06._LEN, LE09)  // _LEN: Length
                 LE09 = 0x24
                 MI09 = 0xFE00B880
                 MA09 = (MI09 + (LE09 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._MIN, MI10)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._MAX, MA10)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._LEN, LE10)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._MIN, MI10)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._MAX, MA10)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y07._LEN, LE10)  // _LEN: Length
                 LE10 = 0x10
                 MI10 = 0xFE00B840
                 MA10 = (MI10 + (LE10 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._MIN, MI11)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._MAX, MA11)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._LEN, LE11)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._MIN, MI11)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._MAX, MA11)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y08._LEN, LE11)  // _LEN: Length
                 LE11 = 0xB4
                 MI11 = 0xFE200000
                 MA11 = (MI11 + (LE11 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._MIN, MI12)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._MAX, MA12)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._LEN, LE12)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._MIN, MI12)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._MAX, MA12)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y09._LEN, LE12)  // _LEN: Length
                 LE12 = 0x20
                 MI12 = 0xFE804000
                 MA12 = (MI12 + (LE12 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._MIN, MI13)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._MAX, MA13)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._LEN, LE13)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._MIN, MI13)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._MAX, MA13)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0A._LEN, LE13)  // _LEN: Length
                 LE13 = 0x20
                 MI13 = 0xFE805000
                 MA13 = (MI13 + (LE13 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._MIN, MI14)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._MAX, MA14)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._LEN, LE14)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._MIN, MI14)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._MAX, MA14)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0B._LEN, LE14)  // _LEN: Length
                 LE14 = 0x20
                 MI14 = 0xFE204000
                 MA14 = (MI14 + (LE14 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._MIN, MI15)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._MAX, MA15)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._LEN, LE15)  // _LEN: Length
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._MIN, MI15)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._MAX, MA15)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0C._LEN, LE15)  // _LEN: Length
                 LE15 = 0x40
                 MI15 = 0xFE215080
                 MA15 = (MI15 + (LE15 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._MIN, MI17)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._MAX, MA17)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._LEN, LE17)  // _LEN: Length
-                LE17 = 0x0100
-                MI17 = 0xFE007000
-                MA17 = (MI17 + (LE17 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._MIN, MI18)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._MAX, MA18)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._LEN, LE18)  // _LEN: Length
-                LE18 = 0x28
-                MI18 = 0xFE20C800
-                MA18 = (MI18 + (LE18 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._MIN, MI19)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._MAX, MA19)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._LEN, LE19)  // _LEN: Length
-                MI19 = 0x7E20C800
-                LE19 = 0x28
-                MA19 = (MI19 + (LE19 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._MIN, MI20)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._MAX, MA20)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._LEN, LE20)  // _LEN: Length
-                MI20 = 0xFF20C000
-                LE20 = 0x28
-                MA20 = (MI20 + (LE20 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._MIN, MI21)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._MAX, MA21)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._LEN, LE21)  // _LEN: Length
-                LE21 = 0x08
-                MI21 = 0xFE1010A0
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._MIN, MI21)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._MAX, MA21)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0D._LEN, LE21)  // _LEN: Length
+                LE21 = 0x0100
+                MI21 = 0xFE007000
                 MA21 = (MI21 + (LE21 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._MIN, MI22)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._MAX, MA22)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._LEN, LE22)  // _LEN: Length
-                LE22 = 0x1000
-                MI22 = 0xFE201000
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._MIN, MI22)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._MAX, MA22)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0E._LEN, LE22)  // _LEN: Length
+                LE22 = 0x28
+                MI22 = 0xFE20C800
                 MA22 = (MI22 + (LE22 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._MIN, MI23)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._MAX, MA23)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._LEN, LE23)  // _LEN: Length
-                LE23 = 0x70
-                MI23 = 0xFE215000
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._MIN, MI23)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._MAX, MA23)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y0F._LEN, LE23)  // _LEN: Length
+                MI23 = 0x7E20C800
+                LE23 = 0x28
                 MA23 = (MI23 + (LE23 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y16._MIN, MI24)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y16._MAX, MA24)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y16._LEN, LE24)  // _LEN: Length
-                LE24 = 0x0100
-                MI24 = 0xFE300000
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._MIN, MI24)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._MAX, MA24)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y10._LEN, LE24)  // _LEN: Length
+                MI24 = 0xC0000000
+                LE24 = One
                 MA24 = (MI24 + (LE24 - One))
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y17._MIN, MI25)  // _MIN: Minimum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y17._MAX, MA25)  // _MAX: Maximum Base Address
-                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y17._LEN, LE25)  // _LEN: Length
-                LE25 = 0x0100
-                MI25 = 0xFE202000
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._MIN, MI25)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._MAX, MA25)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y11._LEN, LE25)  // _LEN: Length
+                LE25 = 0x08
+                MI25 = 0xFE1010A0
                 MA25 = (MI25 + (LE25 - One))
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._MIN, MI26)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._MAX, MA26)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y12._LEN, LE26)  // _LEN: Length
+                LE26 = 0x1000
+                MI26 = 0xFE201000
+                MA26 = (MI26 + (LE26 - One))
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._MIN, MI27)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._MAX, MA27)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y13._LEN, LE27)  // _LEN: Length
+                LE27 = 0x70
+                MI27 = 0xFE215000
+                MA27 = (MI27 + (LE27 - One))
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._MIN, MI28)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._MAX, MA28)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y14._LEN, LE28)  // _LEN: Length
+                LE28 = 0x0100
+                MI28 = 0xFE300000
+                MA28 = (MI28 + (LE28 - One))
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._MIN, MI29)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._MAX, MA29)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y15._LEN, LE29)  // _LEN: Length
+                LE29 = 0x0100
+                MI29 = 0xFE202000
+                MA29 = (MI29 + (LE29 - One))
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y16._MIN, MI30)  // _MIN: Minimum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y16._MAX, MA30)  // _MAX: Maximum Base Address
+                CreateQWordField (RBUF, \_SB.GDV0._CRS._Y16._LEN, LE30)  // _LEN: Length
+                LE30 = 0x0100
+                MI30 = 0xFE007500
+                MA30 = (MI30 + (LE30 - One))
                 Return (RBUF) /* \_SB_.GDV0._CRS.RBUF */
             }
 
             Name (_DMA, ResourceTemplate ()  // _DMA: Direct Memory Access
             {
-                QWordMemory (ResourceConsumer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
+                QWordMemory (ResourceProducer, PosDecode, MinFixed, MaxFixed, NonCacheable, ReadWrite,
                     0x0000000000000000, // Granularity
                     0x00000000C0000000, // Range Minimum
                     0x00000000FFFFFFFF, // Range Maximum
@@ -505,11 +439,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (USB0)
             {
                 Name (_HID, "BCM2848")  // _HID: Hardware ID
-                Name (_CID, Package (0x02)  // _CID: Compatible ID
-                {
-                    "DWC_OTG", 
-                    "DWC2_OTG"
-                })
+                Name (_CID, "BCM2848")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -522,7 +452,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00010000,         // Address Length
-                        _Y18)
+                        _Y17)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x00000069,
@@ -530,7 +460,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.USB0._Y18._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.USB0._Y17._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE980000
                     Return (RBUF) /* \_SB_.GDV0.USB0.RBUF */
                 }
@@ -539,7 +469,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (GPU0)
             {
                 Name (_HID, "BCM2850")  // _HID: Hardware ID
-                Name (_CID, "VC4")  // _CID: Compatible ID
+                Name (_CID, "BCM2850")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -552,7 +482,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00001000,         // Address Length
-                        _Y19)
+                        _Y18)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x0000002A,
@@ -560,11 +490,11 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00006000,         // Address Length
-                        _Y1A)
+                        _Y19)
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000100,         // Address Length
-                        _Y1B)
+                        _Y1A)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x0000004A,
@@ -572,11 +502,11 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000600,         // Address Length
-                        _Y1C)
+                        _Y1B)
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000100,         // Address Length
-                        _Y1D)
+                        _Y1C)
                     I2cSerialBusV2 (0x0050, ControllerInitiated, 0x000186A0,
                         AddressingMode7Bit, "\\_SB.GDV0.I2C2",
                         0x00, ResourceConsumer, , Exclusive,
@@ -588,15 +518,15 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y19._BAS, RB01)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y18._BAS, RB01)  // _BAS: Base Address
                     RB01 = 0xFEC00000
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1A._BAS, RB02)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y19._BAS, RB02)  // _BAS: Base Address
                     RB02 = 0xFE400000
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1B._BAS, RB05)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1A._BAS, RB05)  // _BAS: Base Address
                     RB05 = 0xFE807000
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1C._BAS, RB06)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1B._BAS, RB06)  // _BAS: Base Address
                     RB06 = 0xFE902000
-                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1D._BAS, RB07)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPU0._Y1C._BAS, RB07)  // _BAS: Base Address
                     RB07 = 0xFE808000
                     Return (RBUF) /* \_SB_.GDV0.GPU0.RBUF */
                 }
@@ -643,7 +573,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (RPIQ)
             {
                 Name (_HID, "BCM2849")  // _HID: Hardware ID
-                Name (_CID, "RPIQ")  // _CID: Compatible ID
+                Name (_CID, "BCM2849")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -656,7 +586,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000024,         // Address Length
-                        _Y1E)
+                        _Y1D)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x00000041,
@@ -664,7 +594,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.RPIQ._Y1E._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.RPIQ._Y1D._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE00B880
                     Return (RBUF) /* \_SB_.GDV0.RPIQ.RBUF */
                 }
@@ -673,7 +603,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (VCIQ)
             {
                 Name (_HID, "BCM2835")  // _HID: Hardware ID
-                Name (_CID, "VCIQ")  // _CID: Compatible ID
+                Name (_CID, "BCM2835")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Name (_DEP, Package (0x01)  // _DEP: Dependencies
@@ -690,7 +620,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000010,         // Address Length
-                        _Y1F)
+                        _Y1E)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x00000042,
@@ -698,7 +628,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.VCIQ._Y1F._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.VCIQ._Y1E._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE00B840
                     Return (RBUF) /* \_SB_.GDV0.VCIQ.RBUF */
                 }
@@ -707,7 +637,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (VCSM)
             {
                 Name (_HID, "BCM2856")  // _HID: Hardware ID
-                Name (_CID, "VCSM")  // _CID: Compatible ID
+                Name (_CID, "BCM2856")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Name (_DEP, Package (0x01)  // _DEP: Dependencies
@@ -723,7 +653,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (GPI0)
             {
                 Name (_HID, "BCM2845")  // _HID: Hardware ID
-                Name (_CID, "BCMGPIO")  // _CID: Compatible ID
+                Name (_CID, "BCM2845")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -736,7 +666,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x000000B4,         // Address Length
-                        _Y20)
+                        _Y1F)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x00000091,
@@ -747,7 +677,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.GPI0._Y20._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.GPI0._Y1F._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE200000
                     Return (RBUF) /* \_SB_.GDV0.GPI0.RBUF */
                 }
@@ -756,7 +686,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (I2C1)
             {
                 Name (_HID, "BCM2841")  // _HID: Hardware ID
-                Name (_CID, "BCMI2C")  // _CID: Compatible ID
+                Name (_CID, "BCM2841")  // _CID: Compatible ID
                 Name (_UID, One)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -769,7 +699,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000020,         // Address Length
-                        _Y21)
+                        _Y20)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x00000095,
@@ -783,7 +713,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.I2C1._Y21._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.I2C1._Y20._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE804000
                     Return (RBUF) /* \_SB_.GDV0.I2C1.RBUF */
                 }
@@ -792,7 +722,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (I2C2)
             {
                 Name (_HID, "BCM2841")  // _HID: Hardware ID
-                Name (_CID, "BCMI2C")  // _CID: Compatible ID
+                Name (_CID, "BCM2841")  // _CID: Compatible ID
                 Name (_UID, 0x02)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -805,7 +735,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000020,         // Address Length
-                        _Y22)
+                        _Y21)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x00000095,
@@ -813,7 +743,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.I2C2._Y22._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.I2C2._Y21._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE805000
                     Return (RBUF) /* \_SB_.GDV0.I2C2.RBUF */
                 }
@@ -822,7 +752,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (SPI0)
             {
                 Name (_HID, "BCM2838")  // _HID: Hardware ID
-                Name (_CID, "BCMSPI0")  // _CID: Compatible ID
+                Name (_CID, "BCM2838")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -835,7 +765,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000020,         // Address Length
-                        _Y23)
+                        _Y22)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x00000096,
@@ -860,7 +790,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.SPI0._Y23._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.SPI0._Y22._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE204000
                     Return (RBUF) /* \_SB_.GDV0.SPI0.RBUF */
                 }
@@ -869,7 +799,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (SPI1)
             {
                 Name (_HID, "BCM2839")  // _HID: Hardware ID
-                Name (_CID, "BCMAUXSPI")  // _CID: Compatible ID
+                Name (_CID, "BCM2839")  // _CID: Compatible ID
                 Name (_UID, One)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Name (_DEP, Package (0x01)  // _DEP: Dependencies
@@ -886,7 +816,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000040,         // Address Length
-                        _Y24)
+                        _Y23)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x0000007D,
@@ -906,17 +836,17 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.SPI1._Y24._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.SPI1._Y23._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE215080
                     Return (RBUF) /* \_SB_.GDV0.SPI1.RBUF */
                 }
             }
 
-            Device (PWM0)
+            Device (PWM1)
             {
                 Name (_HID, "BCM2844")  // _HID: Hardware ID
                 Name (_CID, "BCM2844")  // _CID: Compatible ID
-                Name (_UID, Zero)  // _UID: Unique ID
+                Name (_UID, One)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
                 {
@@ -928,18 +858,18 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000100,         // Address Length
-                        _Y25)
+                        _Y24)
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000028,         // Address Length
-                        _Y26)
+                        _Y25)
                     Memory32Fixed (ReadWrite,
-                        0x7E20C800,         // Address Base
+                        0x00000000,         // Address Base
                         0x00000028,         // Address Length
-                        )
+                        _Y26)
                     Memory32Fixed (ReadWrite,
-                        0xFF20C000,         // Address Base
-                        0x00000028,         // Address Length
+                        0xC0000000,         // Address Base
+                        0x00000001,         // Address Length
                         )
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
@@ -953,20 +883,100 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.PWM0._Y25._BAS, RB01)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.PWM1._Y24._BAS, RB01)  // _BAS: Base Address
                     RB01 = 0xFE007000
-                    CreateDWordField (RBUF, \_SB.GDV0.PWM0._Y26._BAS, RB02)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.PWM1._Y25._BAS, RB02)  // _BAS: Base Address
                     RB02 = 0xFE20C800
-                    CreateDWordField (RBUF, \_SB.GDV0.PWM0._Y27._BAS, RB03)  // _BAS: Base Address
-                    RB03 = 0xFE1010A0
-                    Return (RBUF) /* \_SB_.GDV0.PWM0.RBUF */
+                    CreateDWordField (RBUF, \_SB.GDV0.PWM1._Y26._BAS, RB03)  // _BAS: Base Address
+                    RB03 = 0x7E20C800
+                    CreateDWordField (RBUF, \_SB.GDV0.PWM1._Y27._BAS, RB04)  // _BAS: Base Address
+                    RB04 = 0xFE1010A0
+                    Return (RBUF) /* \_SB_.GDV0.PWM1.RBUF */
+                }
+            }
+
+            Device (AAUD)
+            {
+                Name (_HID, "RPI73EA")  // _HID: Hardware ID
+                Name (_DEP, Package (0x01)  // _DEP: Dependencies
+                {
+                    PWM1
+                })
+                Method (_STA, 0, NotSerialized)  // _STA: Status
+                {
+                    Return (0x0F)
+                }
+
+                Name (_DSD, Package (0x02)  // _DSD: Device-Specific Data
+                {
+                    ToUUID ("daffd814-6eba-4d8c-8a91-bc9bbf4aa301") /* Device Properties for _DSD */, 
+                    Package (0x03)
+                    {
+                        Package (0x02)
+                        {
+                            "pwm-clock-source", 
+                            "plld"
+                        }, 
+
+                        Package (0x02)
+                        {
+                            "pwm-clock-source-frequency", 
+                            0x2CB41780
+                        }, 
+
+                        Package (0x02)
+                        {
+                            "pwm-clock-max-frequency", 
+                            0x165A0BC0
+                        }
+                    }
+                })
+            }
+
+            Device (HDMI)
+            {
+                Name (_HID, "RPI73EB")  // _HID: Hardware ID
+                Method (_STA, 0, NotSerialized)  // _STA: Status
+                {
+                    Return (0x0F)
+                }
+
+                Name (RBUF, ResourceTemplate ()
+                {
+                    Memory32Fixed (ReadWrite,
+                        0x00000000,         // Address Base
+                        0x00000100,         // Address Length
+                        _Y28)
+                    Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
+                    {
+                        0x00000075,
+                    }
+                    FixedDMA (0x000A, 0x0005, Width32bit, )
+                })
+                Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
+                {
+                    CreateDWordField (RBUF, \_SB.GDV0.HDMI._Y28._BAS, RB01)  // _BAS: Base Address
+                    RB01 = 0xFE007500
+                    Return (RBUF) /* \_SB_.GDV0.HDMI.RBUF */
+                }
+            }
+
+            Device (HAUD)
+            {
+                Name (_HID, "RPI73EC")  // _HID: Hardware ID
+                Name (_DEP, Package (0x01)  // _DEP: Dependencies
+                {
+                    HDMI
+                })
+                Method (_STA, 0, NotSerialized)  // _STA: Status
+                {
+                    Return (0x0F)
                 }
             }
 
             Device (URT0)
             {
-                Name (_HID, "BCM2837")  // _HID: Hardware ID
-                Name (_CID, "ARMH0011")  // _CID: Compatible ID
+                Name (_HID, "BCM2887")  // _HID: Hardware ID
                 Name (_UID, 0x04)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -979,7 +989,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00001000,         // Address Length
-                        _Y28)
+                        _Y29)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive, ,, )
                     {
                         0x00000099,
@@ -987,18 +997,13 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     PinFunction (Exclusive, PullDown, 0x0007, "\\_SB.GDV0.GPI0", 0x00,
                         ResourceConsumer, ,)
                         {   // Pin list
-                            0x0020,
-                            0x0021
-                        }
-                    PinFunction (Exclusive, PullNone, 0x0006, "\\_SB.GDV0.GPI0", 0x00,
-                        ResourceConsumer, ,)
-                        {   // Pin list
+                            0x001E,
                             0x001F
                         }
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.URT0._Y28._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.URT0._Y29._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE201000
                     Return (RBUF) /* \_SB_.GDV0.URT0.RBUF */
                 }
@@ -1021,7 +1026,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (URTM)
             {
                 Name (_HID, "BCM2836")  // _HID: Hardware ID
-                Name (_CID, "MINIUART")  // _CID: Compatible ID
+                Name (_CID, "BCM2836")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Method (_STA, 0, NotSerialized)  // _STA: Status
@@ -1034,7 +1039,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000070,         // Address Length
-                        _Y29)
+                        _Y2A)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x0000007D,
@@ -1042,7 +1047,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.URTM._Y29._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.URTM._Y2A._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE215000
                     Return (RBUF) /* \_SB_.GDV0.URTM.RBUF */
                 }
@@ -1052,28 +1057,38 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             {
                 Name (_HID, "BCM2EA6")  // _HID: Hardware ID
                 Name (_CID, "BCM2EA6")  // _CID: Compatible ID
+                Name (URIU, One)
                 Method (_STA, 0, NotSerialized)  // _STA: Status
                 {
                     Return (0x0F)
                 }
 
+                Name (BTPL, ResourceTemplate ()
+                {
+                    UartSerialBusV2 (0x0001C200, DataBitsEight, StopBitsOne,
+                        0xC0, LittleEndian, ParityTypeNone, FlowControlHardware,
+                        0x0010, 0x0010, "\\_SB.GDV0.URT0",
+                        0x00, ResourceConsumer, , Exclusive,
+                        )
+                })
+                Name (BTMN, ResourceTemplate ()
+                {
+                    UartSerialBusV2 (0x0001C200, DataBitsEight, StopBitsOne,
+                        0x00, LittleEndian, ParityTypeNone, FlowControlNone,
+                        0x0010, 0x0010, "\\_SB.GDV0.URTM",
+                        0x00, ResourceConsumer, , Exclusive,
+                        )
+                })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    Name (RBUF, ResourceTemplate ()
+                    If ((URIU == Zero))
                     {
-                        UartSerialBusV2 (0x0001C200, DataBitsEight, StopBitsOne,
-                            0x00, LittleEndian, ParityTypeNone, FlowControlNone,
-                            0x0010, 0x0010, "\\_SB.GDV0.URT0",
-                            0x00, ResourceConsumer, , Exclusive,
-                            )
-                        GpioIo (Shared, PullUp, 0x0000, 0x0000, IoRestrictionNone,
-                            "\\_SB.GDV0.RPIQ", 0x00, ResourceConsumer, ,
-                            )
-                            {   // Pin list
-                                0x0080
-                            }
-                    })
-                    Return (RBUF) /* \_SB_.GDV0.BTH0._CRS.RBUF */
+                        Return (BTMN) /* \_SB_.GDV0.BTH0.BTMN */
+                    }
+                    Else
+                    {
+                        Return (BTPL) /* \_SB_.GDV0.BTH0.BTPL */
+                    }
                 }
             }
 
@@ -1227,6 +1242,30 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                         "\\_SB.GDV0.GPI0", 0x00, ResourceConsumer, ,
                         )
                         {   // Pin list
+                            0x000C
+                        }
+                    GpioInt (Edge, ActiveBoth, Shared, PullDown, 0x0000,
+                        "\\_SB.GDV0.GPI0", 0x00, ResourceConsumer, ,
+                        )
+                        {   // Pin list
+                            0x000C
+                        }
+                    GpioIo (Shared, PullDown, 0x0000, 0x0000, IoRestrictionNone,
+                        "\\_SB.GDV0.GPI0", 0x00, ResourceConsumer, ,
+                        )
+                        {   // Pin list
+                            0x000D
+                        }
+                    GpioInt (Edge, ActiveBoth, Shared, PullDown, 0x0000,
+                        "\\_SB.GDV0.GPI0", 0x00, ResourceConsumer, ,
+                        )
+                        {   // Pin list
+                            0x000D
+                        }
+                    GpioIo (Shared, PullDown, 0x0000, 0x0000, IoRestrictionNone,
+                        "\\_SB.GDV0.GPI0", 0x00, ResourceConsumer, ,
+                        )
+                        {   // Pin list
                             0x0010
                         }
                     GpioInt (Edge, ActiveBoth, Shared, PullDown, 0x0000,
@@ -1467,7 +1506,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
             Device (SDC1)
             {
                 Name (_HID, "BCM2847")  // _HID: Hardware ID
-                Name (_CID, "ARASAN")  // _CID: Compatible ID
+                Name (_CID, "BCM2847")  // _CID: Compatible ID
                 Name (_UID, Zero)  // _UID: Unique ID
                 Name (_CCA, Zero)  // _CCA: Cache Coherency Attribute
                 Name (_S1D, One)  // _S1D: S1 Device State
@@ -1484,7 +1523,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                     Memory32Fixed (ReadWrite,
                         0x00000000,         // Address Base
                         0x00000100,         // Address Length
-                        _Y2A)
+                        _Y2B)
                     Interrupt (ResourceConsumer, Level, ActiveHigh, Shared, ,, )
                     {
                         0x0000009E,
@@ -1492,7 +1531,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                 })
                 Method (_CRS, 0, Serialized)  // _CRS: Current Resource Settings
                 {
-                    CreateDWordField (RBUF, \_SB.GDV0.SDC1._Y2A._BAS, RBAS)  // _BAS: Base Address
+                    CreateDWordField (RBUF, \_SB.GDV0.SDC1._Y2B._BAS, RBAS)  // _BAS: Base Address
                     RBAS = 0xFE300000
                     Return (RBUF) /* \_SB_.GDV0.SDC1.RBUF */
                 }
@@ -1505,7 +1544,7 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
                         Package (0x02)
                         {
                             "sdhci-caps", 
-                            0x0100FA81
+                            0x0120FA81
                         }
                     }
                 })
@@ -1621,4 +1660,3 @@ DefinitionBlock ("", "DSDT", 5, "RPIFDN"
         }
     }
 }
-
